### PR TITLE
New Task Suspension Code

### LINKED ISSUE #172: Task Suspension
The current mechanism for suspending and resuming tasks is needlessly confusing, and doesn't allow other tasks to check if a task is suspended or not. This scheme should be replaced by a variable for each task, letting it know if it should be running or not. If not, the task should stop and pend on a semaphore. Otherwise, it'll run as normal. To resume a task, another tasks should set it's variable to the running state and then signal the task's semaphore.

### PR DESCRIPTION
**Pull request by Cam0Cow**
Closes #172

### CODE REVIEW

**Comment by Cam0Cow** [comment]
Doesn't modify SendCarCAN, as it's not yet present in the master branch. This needs to be done before both are merged.

**Comment by Cam0Cow** [comment]
Doesn't modify SendCarCAN, as it's not yet present in the master branch. This needs to be done before both are merged.

**Comment by Cam0Cow** [comment]
Doesn't modify SendCarCAN, as it's not yet present in the master branch. This needs to be done before both are merged.

**Review by ClarkPoon — COMMENTED** [review]
Looks good to me. 

Not approving yet because @Cam0Cow 's comment indicates that some other changes are needed for this to work, and I am guessing this has not been tested

**Comment by babusid** [comment]
@ClarkPoon It shouldn't break SendCarCAN regardless, roie and I talked about it, just gonna do a double verify on the weekend

File: Apps/Src/ArrayConnection.c [hunk 1]
```diff
@@ -62,16 +62,10 @@ void Task_ArrayConnection(void *p_arg) {
 
         if (desiredState == ON && currentState != ON) {
```
Comment (babusid) [inline Apps/Src/ArrayConnection.c:63]: The only task I can think of that activates the array is read-switches and that hasn't been merged in yet. Will this task work without it? I see that we're accessing car_state->ShouldArrayBeActivated, and that's not initialized anywhere.

**Review by babusid — APPROVED** [review]
The only thing that concerns me about this is that the ArrayConnection task depe

**Pull request merged**